DirectGuiDesigner/dialogs/AddItemDialog.py

```diff
@@ -167,35 +181,90 @@ def __callback(self, *args):
 
     def createContent(self):
         for name, value in self.customWidget.addItemExtraArgs.items():
+            valueType = value["type"]
             defaultValue = str(value["defaultValue"])
             self.addLabel(name)
-            self.addEntry(defaultValue)
+            if valueType == "element":
+                self.addElementSelection(name)
+            else:
+                self.addEntry(defaultValue, name)
 
         self.argFrame["canvasSize"] = self.contentBox["frameSize"]
 
     def addLabel(self, text):
         newLabel = DirectLabel(
             scale=LVecBase3f(0.1, 0.1, 0.1),
-            text=text
+            text=text,
+            suppressMouse=True,
+            state=DGG.NORMAL
         )
         self.contentBox.addItem(newLabel)
         self.labelList.append(newLabel)
         self.bindScroll(newLabel)
 
-    def addEntry(self, text=""):
+    def addEntry(self, text="", name=""):
         newEntry = DirectEntry(
             scale=LVecBase3f(0.1, 0.1, 0.1),
             frameColor=(1, 1, 1, 1),
             initialText=text
         )
         self.contentBox.addItem(newEntry)
-        self.entryList.append(newEntry)
+        self.entryList[name] = newEntry
         self.bindScroll(newEntry)
 
+    def addElementSelection(self, name):
+        selection = DirectCollapsibleFrame(
+            frameSize=(0, 1, -1, 0.1),
+            suppressMouse=True,
+            toggleCollapseButton_suppressMouse=True
+        )
+        self.accept(selection.getExtendedEvent(), self.contentBox.refresh)
+        self.accept(selection.getCollapsedEvent(), self.contentBox.refresh)
+        self.bindScroll(selection)
+        self.bindScroll(selection.toggleCollapseButton)
+        self.contentBox.addItem(selection)
+        selectionBox = DirectBoxSizer(
+            orientation='vertical',
+            parent=selection,
+            suppressMouse=True
+        )
+        self.bindScroll(selectionBox)
+        from DirectGuiDesigner.DirectGuiDesigner import DirectGuiDesigner
```

Comment:
Must this import be here? Of prefer to have imports at the top of the file. If it's just for getting the elementDict, we should better pass out in here somehow. To have better control over who uses this data.

Reply:
There is an issue with circular imports if I put if at the top of the file. Perhaps we could add our DirectGuiDesigner instance to base? That way one could use base.DirectGuiDesigner.elementDict instead. 

Otherwise we could just store a reference to DirectGuiDesigner in CusomWidget and AddItemDialog. Or would you prefer that only the elementDict is passed instead of the whole class?

Reply:
Hm, then I'd say passing the elementDict would be the best choice for now except you need more from the DirectGuiDesigner instance itself there. I usually also only passed around the data that was necessary. In general I at least try to not put stuff into global things like base even though I sometimes do it. If passing around the dict would go through too many places to get there, we may should see if the structure can be slimmed down or if not or if it's too complex, then we may put the designer itself into base for easy access.

Reply:
Yes, I do generally agree with that. However, in this case the elementDict would have to be passed around a lot to make this happen. We could simplify the structure a bit, but I think that it would still be inconvenient to pass around the elementDict. 

In my opinion, saving DirectGuiDesigner in base would probably be the cleanest solution.

Reply:
Ok, then we may store that data in base. We should start a new task for collecting possible other locations where this can be used then so we don't have both ways all the time but someday only use the base defined value.
Other than that, I think some diagrams for the current dependency structure and what is used where may be helpful if it's getting more and more complex. Will probably have a look at that at a later time, maybe best if you're done with this topic.